[Docs] Add resources of OpenAI GPT (#20084)
huggingface/transformers: docs/source/en/model_doc/openai-gpt.mdx

If you don't install `ftfy` and `SpaCy`, the [`OpenAIGPTTokenizer`] will default to tokenize
using BERT's `BasicTokenizer` followed by Byte-Pair Encoding (which should be fine for most usage, don't worry).

## Resources

A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with DistilBERT. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.

<PipelineTag pipeline="text-classification"/>

- A blog post on [outperforming OpenAI GPT-3 with SetFit for text-classification](https://www.philschmid.de/getting-started-setfit).

<PipelineTag pipeline="token-classification"/>

> - A course material on [Byte-Pair Encoding tokenization](https://huggingface.co/course/en/chapter6/5)
> - A course material on [training a causal language model from scratch](https://huggingface.co/course/en/chapter7/6)

Reviewer: ```suggestion
- A course material on [Byte-Pair Encoding tokenization](https://huggingface.co/course/en/chapter6/5).
- A course material on [training a causal language model from scratch](https://huggingface.co/course/en/chapter7/6).
```

🚀 Deploy

> - A blog post on how to [deploy BigScience T0_3B to AWS & Amazon SageMaker](https://www.philschmid.de/deploy-bigscience-t0-3b-to-aws-and-amazon-sagemaker)

Reviewer: ```suggestion
- A blog post on how to [deploy BigScience T0_3B to AWS & Amazon SageMaker](https://www.philschmid.de/deploy-bigscience-t0-3b-to-aws-and-amazon-sagemaker).
```

If you don't install `ftfy` and `SpaCy`, the [`OpenAIGPTTokenizer`] will default to tokenize
using BERT's `BasicTokenizer` followed by Byte-Pair Encoding (which should be fine for most usage, don't worry).

## Resources

A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with DistilBERT. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.

<PipelineTag pipeline="text-classification"/>

- A blog post on [outperforming OpenAI GPT-3 with SetFit for text-classification](https://www.philschmid.de/getting-started-setfit).

<PipelineTag pipeline="text-generation"/>

> - OpenAI GPT is supported by this [example script](https://github.com/huggingface/transformers/blob/main/examples/pytorch/text-generation/run_generation.py) and [notebook](https://colab.research.google.com/github/huggingface/blog/blob/main/notebooks/02_how_to_generate.ipynb).

Reviewer: We can use this format to refer to the model directly. Since causal language modeling is also like text generation, you can add that example script here.

Also don't forget to add it's [TensorFlow equivalent](https://huggingface.co/docs/transformers/model_doc/openai-gpt#transformers.TFOpenAIGPTLMHeadModel)!

```suggestion
- [`OpenAIGPTLMHeadModel`] is supported by this [causal language modeling example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling#gpt-2gpt-and-causal-language-modeling), [text generation example script](https://github.com/huggingface/transformers/blob/main/examples/pytorch/text-generation/run_generation.py) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling.ipynb).
```

<PipelineTag pipeline="token-classification"/>

- A course material on [Byte-Pair Encoding tokenization](https://huggingface.co/course/en/chapter6/5).
- A course material on [training a causal language model from scratch](https://huggingface.co/course/en/chapter7/6).

> <PipelineTag pipeline="causal-language-modeling"/>

Reviewer: We can delete this section now since we added causal language modeling under text generation.

- [Causal language modeling](https://huggingface.co/course/en/chapter7/6?fw=pt#training-a-causal-language-model-from-scratch) chapter of the 🤗 Hugging Face Course.
- [`OpenAIGPTLMHeadModel`] is supported by this [causal language modeling example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling#gpt-2gpt-and-causal-language-modeling), [text generation example script](https://github.com/huggingface/transformers/blob/main/examples/pytorch/text-generation/run_generation.py) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling.ipynb).
- [`TFOpenAIGPTLMHeadModel`] is supported by this [causal language modeling example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/language-modeling#run_clmpy) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling-tf.ipynb).

<PipelineTag pipeline="token-classification"/>

- A course material on [Byte-Pair Encoding tokenization](https://huggingface.co/course/en/chapter6/5).

> - A course material on [training a causal language model from scratch](https://huggingface.co/course/en/chapter7/6).

Reviewer: We can remove this since it is the same as the Causal language modeling chapter from the Hugging Face Course linked above.

Author: I deleted the link you mentioned!
